Cargo.toml

```diff
@@ -6,10 +6,15 @@ qcs-api = "0.2.1"
 qcs-api-client-common = "0.6.6"
 qcs-api-client-grpc = "0.6.1"
 qcs-api-client-openapi = "0.7.1"
-quil-rs = "0.16.0"
 serde_json = "1.0.86"
 thiserror = "1.0.37"
 tokio = "1.24.2"
+# We specify quil-rs as a git and versioned dependency so that we can keep the version of
+# quil-rs used in both the Rust and Python packages in sync. The tag used should always
+# be a `quil-py` tag and should match the version used in `crates/python/pyproject.toml`
+# The version must also be specified in order to publish to crates.io. Cargo enforces
+# that the specified version is the same as the version in the git repository.
+quil-rs = { version = "0.17.0", git = "https://github.com/rigetti/quil-rs", tag = "quil-py/v0.1.1" }
```

Comment:
Hmm, what's the issue with just specifying `0.17.0` if it's enforced that the tag's version has to match anyway?

Reply:
I might be misunderstanding the question, but cargo only checks if they match if both are specified. If we specified quil-rs by version only. It would be easier to update it to a version that didn't match the version of quil being used.

For example, In the future, if we release quil-rs `0.18.0` and quil-py `0.2.1`, then when I update the quil-rs version here, cargo will fail until I update the tag to match. If we only specified by version, cargo wouldn't complain at all.